packages/next-urql/src/Provider.ts

```diff
@@ -0,0 +1,25 @@
+'use client';
+
+import React from 'react';
+import { Provider, SSRExchange, Client } from 'urql';
```

Comment:
I haven't changed `createContext` yet to use `import * as React from 'react'; React.createContext`. Should this be done in this same PR? https://github.com/urql-graphql/urql/blob/2107283a7fbe5328641efb67e9784d4d60541dbc/packages/react-urql/src/context.ts#L1

Reply:
Importing contexts is only relevant in server-components, this means that as long as people use the client when they leverage `/rsc` we are good. This is demonstrated in the examples as well, here we create a `React.cached` client and we get it when we want to use it. The good thing here is that in RSC, folks can just use `await` and leverage the data, this can afterwards be passed down to the client over props.